perceval/backends/topicbox/topicbox.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Jose Javier Merchante <jjmerchante@bitergia.com>
+#
+
+import datetime
+import json
+import logging
+
+import dateutil.tz
+
+from grimoirelab_toolkit.datetime import str_to_datetime, datetime_to_utc
+from grimoirelab_toolkit.uris import urijoin
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+from ...client import HttpClient
+
+CATEGORY_MESSAGE = 'message'
+
+MAX_MESSAGES_PER_PAGE = 100
+
+JMAP_CLIENT_CAPABILITIES = [
+    "urn:ietf:params:jmap:mail",
+    "urn:ietf:params:jmap:core",
+    "https://www.topicbox.com/dev/organisation"
+]
+
+DEFAULT_DATETIME = datetime.datetime(1970, 1, 1, 0, 0, 0,
+                                     tzinfo=dateutil.tz.tzutc())
+DEFAULT_LAST_DATETIME = datetime.datetime(2100, 1, 1, 0, 0, 0,
+                                          tzinfo=dateutil.tz.tzutc())
+
+logger = logging.getLogger(__name__)
+
+
+class Topicbox(Backend):
+    """Topicbox backend for Perceval.
+
+    This class retrieves the email messages stored in Topicbox
+    using the JSON Meta Application Protocol (JMAP).
+    Initialize this class passing the URL of the Topicbox application.
+    The origin of the data will be set to the value of `uri`.
+
+    :param uri: URI of the topicbox
+    :param account_id: account id for Topicbox's JMAP
+    :param mailbox_id: mailbox id for Topicbox's JMAP
+    :param tag: label used to mark the data
+    :param archive: archive to store/retrieve items
+    """
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_MESSAGE]
+
+    DATE_FIELD = 'receivedAt'
+    MESSAGE_ID_FIELD = 'id'
+
+    def __init__(self, uri, account_id, mailbox_id, tag=None, archive=None,
+                 max_items=MAX_MESSAGES_PER_PAGE, ssl_verify=True):
+        origin = uri
```

Comment:
If we are retrieving from a mailbox/group, the origin should be the uri to that group. For example, I see on Thunderbird: https://thunderbird.topicbox.com/groups/beta which is the group of ThunderBird Beta